### ISSUE TITLE
react-scripts Build error during creation of optimized production build

### ISSUE THREAD

**Issue by hasnogame** [drawpile/dpserver#4]
I cannot get this script to create a working Docker container.

Environment: Fresh install of a Debian 10 headless-server on a publicly-available VPS with a domain name. I have also tried my self-owned Docker setup running on Ubuntu 22.04 with the same result. Let's Encrypt certificate is created and tested functional.

```
root@server:~/dpserver# ./setup.sh
Reading package lists... Done
Building dependency tree
Reading state information... Done
apt-transport-https is already the newest version (1.8.2.3).
ca-certificates is already the newest version (20200601~deb10u2).
ca-certificates set to manually installed.
gnupg2 is already the newest version (2.2.12-1+deb10u2).
software-properties-common is already the newest version (0.96.20.2-2).
curl is already the newest version (7.64.0-4+deb10u3).
0 upgraded, 0 newly installed, 0 to remove and 121 not upgraded.
OK
Hit:1 http://security.debian.org/debian-security buster/updates InRelease
Hit:2 http://deb.debian.org/debian buster InRelease
Hit:3 http://deb.debian.org/debian buster-updates InRelease
Hit:4 https://download.docker.com/linux/debian buster InRelease
Reading package lists... Done
Building dependency tree
Reading state information... Done
121 packages can be upgraded. Run 'apt list --upgradable' to see them.
Reading package lists... Done
Building dependency tree
Reading state information... Done
docker-compose is already the newest version (1.21.0-3).
docker-ce is already the newest version (5:20.10.22~3-0~debian-buster).
0 upgraded, 0 newly installed, 0 to remove and 121 not upgraded.
Reading package lists... Done
Building dependency tree
Reading state information... Done
apache2-utils is already the newest version (2.4.38-3+deb10u8).
dialog is already the newest version (1.3-20190211-1).
git is already the newest version (1:2.20.1-2+deb10u6).
0 upgraded, 0 newly installed, 0 to remove and 121 not upgraded.
Already up to date.

added 1 package, and audited 1495 packages in 29s

6 packages are looking for funding
  run `npm fund` for details

86 vulnerabilities (1 low, 27 moderate, 44 high, 14 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
npm notice
npm notice New major version of npm available! 8.19.2 -> 9.2.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v9.2.0
npm notice Run npm install -g npm@9.2.0 to update!
npm notice

> dpwebadmin@0.1.0 build
> react-scripts build

Creating an optimized production build...
Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:133:10)
    at module.exports (/app/node_modules/webpack/lib/util/createHash.js:135:53)
    at NormalModule._initBuildHash (/app/node_modules/webpack/lib/NormalModule.js:417:16)
    at handleParseError (/app/node_modules/webpack/lib/NormalModule.js:471:10)
    at /app/node_modules/webpack/lib/NormalModule.js:503:5
    at /app/node_modules/webpack/lib/NormalModule.js:358:12
    at /app/node_modules/loader-runner/lib/LoaderRunner.js:373:3
    at iterateNormalLoaders (/app/node_modules/loader-runner/lib/LoaderRunner.js:214:10)
    at iterateNormalLoaders (/app/node_modules/loader-runner/lib/LoaderRunner.js:221:10)
    at /app/node_modules/loader-runner/lib/LoaderRunner.js:236:3
    at runSyncOrAsync (/app/node_modules/loader-runner/lib/LoaderRunner.js:130:11)
    at iterateNormalLoaders (/app/node_modules/loader-runner/lib/LoaderRunner.js:232:2)
    at Array.<anonymous> (/app/node_modules/loader-runner/lib/LoaderRunner.js:205:4)
    at Storage.finished (/app/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:55:16)
    at /app/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:91:9
/app/node_modules/react-scripts/scripts/build.js:19
  throw err;
  ^

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:133:10)
    at module.exports (/app/node_modules/webpack/lib/util/createHash.js:135:53)
    at NormalModule._initBuildHash (/app/node_modules/webpack/lib/NormalModule.js:417:16)
    at /app/node_modules/webpack/lib/NormalModule.js:452:10
    at /app/node_modules/webpack/lib/NormalModule.js:323:13
    at /app/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /app/node_modules/loader-runner/lib/LoaderRunner.js:233:18
    at context.callback (/app/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at /app/node_modules/babel-loader/lib/index.js:55:103
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

Node.js v18.12.1
```

I can still run the container with `docker-compose up -d` but when attempting to access the server's GUI using `drawpile-srv.exe --gui --remote https://[server domain]` this is my output in CMD:

```
c:\Program Files\Drawpile>drawpile-srv.exe --gui --remote https://[server domain]
Creating new NetworkAccessManager for thread QThread(0x16471d0)
Switching from page "" to "summary:server"
"serversettings" query failed: QJsonDocument()
"sessionlist" query failed: QJsonDocument()
"sessionlist" query failed: QJsonDocument()
Switching from page "summary:server" to "sessionlist"
Switching from page "sessionlist" to "banlist"
"banlist" query failed: QJsonDocument()
"sessionlist" query failed: QJsonDocument()
Switching from page "banlist" to "accountlist"
"accountlist" query failed: QJsonDocument()
"sessionlist" query failed: QJsonDocument()
```

On the server side of things this message repeats while I have the connection open between the GUI and the headless server:

```
nginx_1         | [my ip] - - [03/Jan/2023:15:04:02 +0000] "GET /sessions HTTP/1.1" 404 153 "-" "Mozilla/5.0" "-"
nginx_1         | [my ip] - - [03/Jan/2023:15:04:07 +0000] "GET /sessions HTTP/1.1" 404 153 "-" "Mozilla/5.0" "-"
nginx_1         | [my ip] - - [03/Jan/2023:15:04:12 +0000] "GET /sessions HTTP/1.1" 404 153 "-" "Mozilla/5.0" "-"
nginx_1         | [my ip] - - [03/Jan/2023:15:04:17 +0000] "GET /sessions HTTP/1.1" 404 153 "-" "Mozilla/5.0" "-"
nginx_1         | [my ip] - - [03/Jan/2023:15:04:22 +0000] "GET /sessions HTTP/1.1" 404 153 "-" "Mozilla/5.0" "-"
nginx_1         | [my ip] - - [03/Jan/2023:15:04:27 +0000] "GET /sessions HTTP/1.1" 404 153 "-" "Mozilla/5.0" "-"
nginx_1         | [my ip] - - [03/Jan/2023:15:04:32 +0000] "GET /sessions HTTP/1.1" 404 153 "-" "Mozilla/5.0" "-"
nginx_1         | [my ip] - - [03/Jan/2023:15:04:37 +0000] "GET /sessions HTTP/1.1" 404 153 "-" "Mozilla/5.0" "-"
```

The Drawpile server is functional and hosting sessions works without issue. It is only the GUI access which does not work at all. Any assistance would be appreciated!

**Comment by askmeaboutlo0m** [comment]
Fixed in #5. Until that's merged, you can clone <https://github.com/askmeaboutlo0m/dpserver> instead or patch the change in yourself, it's only one line.

**Comment by askmeaboutlo0m** [comment]
That's merged now, so closing as resolved.
